mule-linter-core/src/main/groovy/com/avioconsulting/mule/linter/rule/configuration/AutoDiscoveryRule.groovy

```diff
@@ -0,0 +1,146 @@
+     * DEFAULT_API_IDS: is a default list of values for API IDs in the property files that will be replaced at deployment time.
+     * API Auto Discovery rule will not check property value rules when apiId value matches in the DEFAULT_API_IDs list.
+     * The default list is: ["-1", "0", "1"]
+     */
+    private static List DEFAULT_API_IDS = ["-1", "0", "1"]
+
+    /**
+     * environments: is a list of environments that the property must be found in.
+     * This value is used when determining the name for property files to be searched.
+     * The default list is: ['dev', 'test', 'prod']
+     */
+    @Param("environments") List<String> environments
+
+    /**
+     * pattern: is a custom naming scheme for property files loaded by environment.
+     * The default pattern is `"${appname}-${env}.properties"`.
+     */
+    @Param("pattern") String pattern
+
+    AutoDiscoveryRule() {
+        super(RULE_ID, RULE_NAME, RuleSeverity.BLOCKER, RuleType.VULNERABILITY)
+        this.environments = DEFAULT_ENVIRONMENTS
+        this.pattern = DEFAULT_PATTERN
+    }
+    @Override
+    List<RuleViolation> execute(Application app) {
+        List<RuleViolation> violations = []
+        List<String> httpFlowsList = getHttpListenerFlows(app)
+        List<MuleComponent> autoDiscoveryComponents = app.findComponents(AUTO_DISCOVERY_COMPONENT, AUTO_DISCOVERY_NAMESPACE)
+        httpFlowsList.each { httpFlow ->
+            MuleComponent autoDiscoveryComponent = findMatchingAutoDiscoveryComponent(httpFlow, autoDiscoveryComponents)
+            if (autoDiscoveryComponent != null) {
+                // Check if apiId property exists for API Autodiscovery component
+                if (!autoDiscoveryComponent.hasAttributeValue("apiId") || autoDiscoveryComponent.getAttributeValue("apiId").empty) {
+                    violations.add(new RuleViolation(this, autoDiscoveryComponent.file.path,
+                            autoDiscoveryComponent.lineNumber, API_ID_MISSING_VIOLATION_MESSAGE))
+                } else {
+                    String apiIdAttributeValue = autoDiscoveryComponent.getAttributeValue("apiId")
+                    // Check if apiId property is externalized for API Autodiscovery, and not hardcoded
+                    if (apiIdAttributeValue.startsWith('${')) {
+                        String apiIdProperty = apiIdAttributeValue.substring(2, apiIdAttributeValue.length() - 1)
+                        violations.addAll checkPropertyValueViolations(app, apiIdProperty)
+                    } else {
+                        violations.add(new RuleViolation(this, autoDiscoveryComponent.file.path,
+                                autoDiscoveryComponent.lineNumber, API_ID_HARDCODED_VIOLATION_MESSAGE+" The Hard coded value for API ID is : "+apiIdAttributeValue))
+                    }
+                }
+            } else {
+                violations.add(new RuleViolation(this, httpFlow,0, MISSING_AUTODISCOVERY_MESSAGE + httpFlow))
+            }
```

Comment:
too much if-else nesting. maybe doing a braking down in methods it'll better for readability.

Reply:
Updated these conditions into separate methods to make more readable now..